### ISSUE TITLE
 Resizing clients with mouse draws them at incorrect positions briefly

### ISSUE THREAD

**Issue by YaLTeR** [swaywm/sway#3636]
* Sway Version: `1.0-rc1-38-g995b6eda (Feb 10 2019, branch 'master')`
* Debug Log: https://gist.github.com/YaLTeR/30828bf0583861debf061116d55e3016
* Configuration File: default

Open two tiling clients (I chose alacritty in the session with the log) and resize them quickly with mouse. Alternatively open a floating client and resize that (easier to see when resizing the left part of the client).

**Comment by c-edw** [comment]
I noticed this a while ago. Not sure if it's fixable, but I could be wrong.

**Comment by YaLTeR** [comment]
FWIW it seems to work fine on rootston.

**Comment by c-edw** [comment]
Yeah that's a good indicator that it can be fixed then. I'll see if I can find the cause.

**Comment by c-edw** [comment]
I think it could be here. Rootston's version of this function looks a bit more advanced and does some math to apply damage more accurately.

https://github.com/swaywm/sway/blob/fb4f29289fd8d1f94975e64fded6657222ed5390/sway/desktop/layer_shell.c#L264-L287

**Comment by emersion** [comment]
This is the layer shell part, which is very unlikely to be related to this issue.

Does it happen at all when resizing from the right/bottom side?

**Comment by YaLTeR** [comment]
Yes, it happens, just with a slightly lower "incorrect position" amplitude.

**Comment by YaLTeR** [comment]
Worth noting that currently (`1.0-rc1-149-g7f700e08 (Mar  9 2019, branch 'master')`) this is almost completely not reproducible on `alacritty` (although it still happens), but still very well visible on something like `nautilus` (or when tiling `alacritty` and `nautilus`). It also leaves broken pixels behind outside if you play around with it, although they seem to disappear on any further interaction.

**Comment by l4rzy** [comment]
I don't know it this is related but it seems to be happened with `gtk-demo` applications too.
sway version: `1.1-rc1-3-g9dca18b6 (May  7 2019, branch 'master')`
and it did not happen is a short amount of time, but it stays like that permanently
![img](https://i.imgur.com/kcd7YG2.png)
and mpv as well
![img](https://i.imgur.com/46Qo2Ej.png)

**Comment by Tromino** [comment]
It seems like the position of the window on-screen is being updated to center itself according to what its size and position is supposed to be, before the client actually gets a chance to resize itself. I'd guess it's probably related to the changes in #3523 in some way (it would make this issue more visible, at least).

A good way of triggering this seems to be to open 3 windows (with 2 of them vertically tiled in a container) and resize one of the windows from its corner. This causes the clients' sizes to update very slowly on both machines I've tried it on, and makes the effect of this bug very visible.

**Comment by kennylevinsen** [comment]
Resize issues might be fixed by https://github.com/swaywm/sway/pull/5403 (not content-clipping, that's separate).
